### PR TITLE
Wagtail CMS changes for bootcamp certificate

### PR DESCRIPTION
**Pull request by asadiqbal08**
#### What are the relevant tickets?
#1101 

#### What's this PR do?
Integrated the certificate and signatories module in wagtail CMS

#### How should this be manually tested?
Try to create a generic certificate page in wagtail under bootcamp index page.

### CODE REVIEW

**Comment by asadiqbal08** [comment]
Guys, Please let me know if any other field is required in the certificate or signatory models ?

File: cms/models.py [hunk 1]
```diff
@@ -736,3 +740,195 @@ class ResourcePagesSettings(BaseSetting):
...
+    promote_panels = []
+    parent_page_types = [SignatoryIndexPage]
+    subpage_types = []
+
+    name = models.CharField(
+        max_length=250, null=False, blank=False, help_text="Name of the signatory."
+    )
+    title_1 = models.CharField(
+        max_length=250,
+        null=True,
```
Comment (HamzaIbnFarooq) [inline cms/models.py:798]: FYI: It's a good practice to not use `null=True` for Char or Text Fields.
https://simpleisbetterthancomplex.com/tips/2016/07/25/django-tip-8-blank-or-null.html

File: cms/models.py [hunk 2]
```diff
@@ -736,3 +740,195 @@ class ResourcePagesSettings(BaseSetting):
...
+
+    def serve(self, request, *args, **kwargs):
+        """
+        As the name suggests these pages are going to be children of some other page. They are not
+        designed to be viewed on their own so we raise a 404 if someone tries to access their slug.
+        """
+        raise Http404
+
+
+class CertificatePage(BootcampRunChildPage):
```
Comment (HamzaIbnFarooq) [inline cms/models.py:851]: @asadiqbal08 do you think that we need CertificateIndexPage for certificate routing under HomePage, like we did in xPro?

File: cms/templates/certificate_page.html [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:site" content="@MITxonedX" />
+    <meta name="twitter:image" content="{{ share_image_url }}" />
+    <meta property="og:title" content="{{ site_name }} | Certificate for: {{ page.product_name }}" />
+    <meta property="og:description" content="Certificate for {{ page.product_name }} awarded by {{ site_name }}." />
+    <meta property="og:url" content="{{ request.build_absolute_uri }}" />
+    <meta property="og:image:url" content="{{ share_image_url }}" />
+    <meta property="og:image:width" content="{{ share_image_width }}" />
+    <meta property="og:image:height" content="{{ share_image_height }}" />
+    <meta property="og:image:alt" content="A certificate from MIT xPRO" />
```
Comment (HamzaIbnFarooq) [inline cms/templates/certificate_page.html:19]: Currently, this template will not work as we haven't passed any context to this template as well as this template needs tweaks according to bootcamps.

**Review by HamzaIbnFarooq — CHANGES_REQUESTED** [review]
some minor questions and some of the tests are also failing.

File: cms/models.py (hunk 2 above)
Comment (asadiqbal08) [reply cms/models.py:851]: I guess, Yes we need it.

**Review by HamzaIbnFarooq — APPROVED** [review]
LGTM

**Comment by pdpinch** [comment]
I know this has already merged, but I'd like to see some screenshots.

**Comment by asadiqbal08** [comment]
@pdpinch sequence of screens are here. Please feel free to share thoughts about the certificate fields. 

Screen Shot 2020-12-18 at 12 33 25 AM

Screen Shot 2020-12-18 at 12 33 37 AM

Screen Shot 2020-12-18 at 12 33 44 AM


screencapture-bc-odl-local-8099-cms-pages-add-cms-certificatepage-11-2020-12-18-00_33_49

**Comment by pdpinch** [comment]
Thanks Asad. I have a few questions: 

1. How is the Certificate related to the Bootcamp Run? Is that implied by the parent/child relationship, or do we need an explicit field that relates them?
2. The Subtitle should be optional. Also, I think we should relabel some of these fields (see below) 
3. I should have thought of this, but we'll need some content hints.  

How about this:

- 'Bootcamp Run Name, e.g. "MIT Innovation and Entrepreneruship Bootcamp"'
- 'Certificate Name, e.g. "Certificate in New Ventures Leadership"'
- 'Location, e.g. "Brisbane, Australia"'

**Comment by asadiqbal08** [comment]
@pdpinch I believe, Certificate will relate to bootcamp run in the same way as did in xPro, Parent child relationship. 
and also can you please take a look over this PR https://github.com/mitodl/bootcamp-ecommerce/pull/1108 ?

**Pull request merged**